lib/miq_automation_engine/engine/miq_ae_object.rb

```diff
@@ -279,11 +279,15 @@ def process_args_attribute(args, args_key)
     end
 
     def load_array_objects_from_string(objects_str)
-      objects_str.split(',').collect do |o|
-        klass, str_value = o.split(CLASS_SEPARATOR)
-        value = MiqAeObject.convert_value_based_on_datatype(str_value, klass)
-        value if value.kind_of?(MiqAeMethodService::MiqAeServiceModelBase)
-      end.compact
+      objects_str.split(',').collect do |element|
+        if element.include?(CLASS_SEPARATOR)
+          klass, str_value = element.split(CLASS_SEPARATOR)
+          MiqAeObject.convert_value_based_on_datatype(str_value.strip, klass.strip)
+        else
+          element.presence
+        end.compact
+      end
```

Comment:
@syncrou The `.compact` belongs on the `end` for the `collect` block.  It just need to move down one line, and you can remove the empty line as well.  This should resolve most of the failing tests.